src/pip/_internal/req/constructors.py

```diff
@@ -74,6 +74,23 @@ def _strip_extras(path):
     return path_no_extras, extras
 
 
+def convert_extras(extras):
```

Comment:
determine extras?

Reply:
Determine makes it sound like it should do more IMO.

Reply:
Naming things is difficult. 🙃

I'm happy to let you merge this as is. I guess if you can come up with a better name here (explicit is OK), that'd be nice.

Reply:
If it's OK I think we should merge #6203 first, then I can work this in on top of it.